### PR TITLE
Upgrade to thrift 0.17 and fix issues

### LINKED ISSUE #3105: Upgrade to thrift 0.17
**Is your feature request related to a problem or challenge? Please describe what you are trying to do.**

Upgrade to thrift 0.17

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by Jimexist**
# Which issue does this PR close?

- Closes #3105
- update parquet.rs

# Rationale for this change
 

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

File: parquet/Cargo.toml [hunk 1]
```diff
@@ -41,7 +41,7 @@ arrow-ipc = { version = "27.0.0", path = "../arrow-ipc", default-features = fals
 
 ahash = { version = "0.8", default-features = false, features = ["compile-time-rng"] }
 bytes = { version = "1.1", default-features = false, features = ["std"] }
-thrift = { version = "0.16", default-features = false }
+thrift = { version = "0.17", package = "databend-thrift", default-features = false }
```
Comment (Jimexist) [inline parquet/Cargo.toml:44]: will change back once verification is done. we are in the process of publishing this.

**Comment by Jimexist** [comment]
`cargo doc` failed, not sure we should just ignore the format file which is autogenerated

**Comment by tustvold** [comment]
> cargo doc failed, not sure we should just ignore the format file which is autogenerated

My vote would be to manually patch up the doc comments it is complaining about. That's what I did before... Might even be scriptable...

**Review by tustvold — APPROVED** [review]
I've added API change as I presume this makes breaking changes to the format module. I think once the docs are fixed this can go in, thank you :+1:

**Pull request merged**